packages/build/src/core/constants.ts

```diff
@@ -53,9 +128,8 @@ export const getConstants = async function ({
     // The directory where internal Edge Functions (i.e. generated programmatically
     // via plugins or others) live
     INTERNAL_EDGE_FUNCTIONS_SRC: join(buildDir, packagePath || '', INTERNAL_EDGE_FUNCTIONS_SRC),
-  }
-  const constantsA = await addMutableConstants({ constants, buildDir, netlifyConfig })
-  return constantsA
+  } as const
+  return (await addMutableConstants({ constants, buildDir, netlifyConfig })) as unknown as NetlifyPluginConstants
```

Comment:
Why are you awaiting this?

Reply:
just to avoid having the `Promise<>` on the type.

It should not change much except keep the error context on this file instead of passing it up to the next function as we return a promise.

This is a pattern we are following among netlify/build to await returning promises to keep the error context scoped to these functions.